exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -386,17 +387,16 @@ public static void checkBackPaths(String parent, String combinedPath, String sub
     Preconditions.checkArgument(!combinedPath.isEmpty(), "Empty path (" + combinedPath + "( in file selection path.");
 
     if (!combinedPath.startsWith(parent)) {
-      StringBuilder msg = new StringBuilder();
-      msg.append("Invalid path : ").append(subpath).append(" takes you outside the workspace.");
-      throw new IllegalArgumentException(msg.toString());
+      throw new IllegalArgumentException(
+        String.format("Invalid path [%s] takes you outside the workspace.", subpath));
     }
   }
 
   public List<FileStatus> getFileStatuses() {
     return statuses;
   }
 
-  public boolean supportDirPrunig() {
+  public boolean supportDirPruning() {
```

Comment:
Good catch. `suppportsDirPruning` (with an s)?

The `support` form is imperative, it tells this object to support dir pruning. The `supports` form asks if this object does or does not support dir pruning.